### PR TITLE
Feature/new inputs in session joining (#75) (#168)

### PR DESCRIPTION
**Pull request by Derstilon**
New and improved form with conditional input of session key invitation.
image
I Will add a QR scanner later 
Text fields now have content verification. Session invitation can only be a number with 7 digits. Name and Surname can be multiple words each of them starts with uppercase forced by code.

I've also made a crucial bug fix for local storage that was breaking the app. 😤

### CODE REVIEW

File: src/student/components/joinSessionView/StudentFormView.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    const handleChangeName = (event: ChangeEvent<HTMLInputElement>) => {
+        let nameInput = "";
+        event.target.value.toLowerCase().replace(/[^a-zA-ZąęłżźćóńśĄŻŹĆĘŁÓŃŚ ]/gi, "").split(" ").forEach(word => {
+            if(nameInput.length != 0)
+                nameInput += " ";
+            if(word.length > 0)
+                nameInput += word[0].toUpperCase() + word.substring(1);
+            else
+                nameInput += word;
+        });
```
Comment (ostatni5) [inline src/student/components/joinSessionView/StudentFormView.tsx:80]: Code duplication(1)

File: src/student/components/joinSessionView/StudentFormView.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    const handleChangeSurname = (event: ChangeEvent<HTMLInputElement>) => {
+        let surnameInput = "";
+        event.target.value.toLowerCase().replace(/[^a-zA-ZąęłżźćóńśĄŻŹĆĘŁÓŃŚ ]/gi, "").split(" ").forEach(word => {
+            if(surnameInput.length != 0)
+                surnameInput += " ";
+            if(word.length > 0)
+                surnameInput += word[0].toUpperCase() + word.substring(1);
+            else
+                surnameInput += word;
+        });
```
Comment (ostatni5) [inline src/student/components/joinSessionView/StudentFormView.tsx:92]: Code duplication(1)

File: src/student/components/joinSessionView/StudentFormView.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            };
+
+            if (session)
+                backEnd
+                    ?.joinLecture(session, fakeStudent)
+                    .then((response) => {
+                        console.log(response);
+                        store.studentNick = fakeStudent.nick;
+                        store.invitation = session;
+                        store.studentId = "1";//response.student_id;
```
Comment (ostatni5) [inline src/student/components/joinSessionView/StudentFormView.tsx:117]: ` store.studentId = "1"; //response.student_id;` are you sure about that ?

**Review by ostatni5 — CHANGES_REQUESTED** [review]
Good job...
obraz

**Review by ostatni5 — APPROVED** [review]
It's not like I like your code or something...
obraz

**Review by rozchlastywacz — APPROVED** [review]
whenever i see your code:
c55e5d7dca5c7a8e80d250e45593069b

**Pull request merged**